python/ray/serve/_private/deployment_state.py

```diff
@@ -1308,28 +1313,43 @@ def _scale_deployment_replicas(self) -> bool:
             )
             to_add = max(delta_replicas - stopping_replicas, 0)
             if to_add > 0:
+                # Exponential backoff
+                failed_to_start_threshold = min(
+                    MAX_DEPLOYMENT_CONSTRUCTOR_RETRY_COUNT,
+                    self._target_state.num_replicas * 3,
```

Comment:
What's the reason for choosing `self._target_state.num_replicas * 3`?

Reply:
Ah is it to compare against the total number of replica restarts across the deployment? (So on average each replica has failed 3 times?)

Reply:
I used the threshold used for setting the deployment unhealthy: [code pointer](https://github.com/ray-project/ray/blob/master/python/ray/serve/_private/deployment_state.py#L1388). Basically, perform exponential backoff after a replica fails 3 times and the deployment is determined unhealthy.

Reply:
> Ah is it to compare against the total number of replica restarts across the deployment? (So on average each replica has failed 3 times?)

Yup, I believe so.